### ISSUE TITLE
[ONNX] Rename dynamic shapes produced by ExportedProgram to dynamic_axes

### ISSUE THREAD

**Issue by titaiwangms** [pytorch/pytorch#143443]
`torch.export.export` names dynamic shapes to be s0, s1, s2, s3, ..., etc. However, in ONNX, users could pass in the naming through `dynamic_axes` and `input_names`. We need to rename them to what users request.

**Comment by titaiwangms** [comment]
Was demo on an example:

```python
    class Model(torch.nn.Module):
        def forward(self, x, y):
            return x + y
    
    dim = torch.export.Dim("x", min=1, max=6)
    onnx_program = torch.onnx.export(
        Model(),
        (torch.randn(2, 3), torch.randn(2, 3)),
        input_names=["custom_input_x", "custom_input_y"],
        dynamic_shapes=[{0: dim}, {0: dim}],
        dynamo=True,
    )
```
and got ONNX model like the following in IR:

```
    graph=Graph(
        name='main_graph',
        inputs=(
            %"custom_input_x"<FLOAT,[s0,3]>,
            %"custom_input_y"<FLOAT,[s0,3]>
        ),
        outputs=(
            %"add"<FLOAT,[s0,3]>
        ),
        len()=1
    )
```

It looks like what we want is actually controlled by `torch.export.Dim`. We set the dim to 'x', but the axes still show as s0. I think this might be better addressed from export team side.

cc @xadupre @gramalingam @justinchuby

**Comment by titaiwangms** [comment]
https://github.com/pytorch/pytorch/issues/144273
